keras_cv/layers/preprocessing/random_rotation_test.py

```diff
@@ -0,0 +1,122 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.random_rotation import RandomRotation
+
+
+class RandomRotationTest(tf.test.TestCase):
+    def test_random_rotation_output_shapes(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomRotation(0.5)
+        actual_output = layer(input_images, training=True)
+        self.assertEqual(expected_output.shape, actual_output.shape)
+
+    def test_random_rotation_inference(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomRotation(0.5)
+        actual_output = layer(input_images, training=False)
+        self.assertAllClose(expected_output, actual_output)
+
+    def test_distribution_strategy(self):
+        """Tests that RandomRotation can be created within distribution
+        strategies."""
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        strat = tf.distribute.MirroredStrategy(devices=["cpu", "gpu"])
+        with strat.scope():
+            layer = RandomRotation(0.5)
+            output = strat.run(lambda: layer(input_images, training=True))
+        values = output.values
+        self.assertAllEqual(2, len(values))
+
+    def test_config_with_custom_name(self):
+        layer = RandomRotation(0.5, name="image_preproc")
+        config = layer.get_config()
+        layer_1 = RandomRotation.from_config(config)
+        self.assertEqual(layer_1.name, layer.name)
+
+    def test_unbatched_image(self):
+        input_image = np.reshape(np.arange(0, 25), (5, 5, 1)).astype(np.float32)
+        # 180 rotation.
+        layer = RandomRotation(factor=(0.5, 0.5))
+        output_image = layer(input_image)
+        expected_output = np.asarray(
+            [
+                [24, 23, 22, 21, 20],
+                [19, 18, 17, 16, 15],
+                [14, 13, 12, 11, 10],
+                [9, 8, 7, 6, 5],
+                [4, 3, 2, 1, 0],
+            ]
+        ).astype(np.float32)
+        expected_output = np.reshape(expected_output, (5, 5, 1))
+        self.assertAllClose(expected_output, output_image)
+
+    def test_augment_image(self):
+        input_image = np.reshape(np.arange(0, 25), (5, 5, 1)).astype(np.float32)
+        # 180 rotation.
+        layer = RandomRotation(factor=(0.5, 0.5))
+        output_image = layer.augment_image(
+            input_image, transformation=layer.get_random_transformation()
```

Comment:
Instead of get random transformation can we call the layer?

Reply:
@luke this test is specifically to verify augment_image(). So I am testing that separately.

Reply:
Ideally, tests should be structured to use user facing APIs instead of relying on implementation details.  Call will do the same thing here, and if it does not that is symptomatic of a greater error at play

Reply:
Cool, In that case the tests are already covered. I will remove this.